### PR TITLE
Add Option::xor method

### PR DESCRIPTION
**Pull request by clarfonthey**
Implements the method requested in #50512.

### CODE REVIEW

**Comment by ishitatsuyuki** [comment]
Please go over the RFC process before implementing a new library function.

**Comment by Mark-Simulacrum** [comment]
The RFC process is not generally necessary for adding unstable library features; approval from the libs team is sufficient.

**Review by andre-vm — CHANGES_REQUESTED** [review]

File: src/libcore/option.rs [hunk 1]
```diff
@@ -705,6 +705,42 @@ impl<T> Option<T> {
...
+    /// ```
+    /// #![feature(option_xor)]
+    ///
+    /// let x = Some(2);
+    /// let y: Option<u32> = None;
+    /// assert_eq!(x.xor(y), Some(2));
+    ///
+    /// let x: Option<u32> = None;
+    /// let y = Some(2);
+    /// assert_eq!(x.xor(y), None);
```
[the remark this answers is not in the document]
Comment (clarcharr) [reply src/libcore/option.rs]: Oops, fixed.

**Comment by sfackler** [comment]
Seems plausible for me.

cc @rust-lang/libs 

@bors r+

**Pull request merged**